app/src/components/UsersDashboard/UsersDashboard.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { ViewState } from "../ToggleView/ToggleViewState";
+import { User } from "../../api/usersApi.ts";
+import CreatorCard from "../CreatorCard/CreatorCard.tsx";
+import Row from "../TableRow/Row/Row.tsx";
+import TableHeadRow from "../TableRow/TableHeadRow/TableHeadRow.tsx";
+import "./UsersDashboard.scss";
+
+type UsersDashboardProps = {
+    viewState: ViewState;
+    users: User[];
+    onUserClick: (user: User) => void;
+}
+
+function UsersDashboard(props: UsersDashboardProps) {
+    switch (props.viewState) {
+        case 'grid':
+            return <GridUsers {...props} />;
+        case 'list':
+            return <ListUsers {...props} />;
+    }
```

Comment:
The `props.viewState` isn't utilized by the `GridUsers` `ListUser`, so it's unnecessary to pass it. 

General rule of thumb: pass the minimal required information into functions / components.